test/e2e/shared/common.go

```diff
@@ -176,6 +177,11 @@ func dumpMachine(ctx context.Context, e2eCtx *E2EContext, machine infrav1.OpenSt
 		_, _ = fmt.Fprintf(GinkgoWriter, "error writing server JSON %s: %s", serverJSON, err)
 	}
 
+	srvUser := "ubuntu"
+	if strings.Contains(machine.Spec.Image, "flatcar") {
+		srvUser = "core"
+	}
```

Comment:
I wonder if it would make sense to add this to the e2e_conf.yaml? We could set it to "ubuntu" there and then override for the flatcar test. Not sure if it would work though as this is in the common.go, just a thought.